File: 009-Move Hyphens to Front/MoveHyphenstoFront.cpp
#include<iostream>
#include<cstring>
using namespace std;

char* MoveHyphen(char str[], int n){
    if(str == nullptr || n <= 0){
        return nullptr;
    }
    string HypenPart = "";
    string NonHypenPart = "";   
    for(int i = 0; i < n; i++)
    {
        if(str[i] == '-')
        {
            HypenPart += str[i];
        }else
        {
            NonHypenPart += str[i];
        }
    }

    string result = HypenPart + NonHypenPart;
    strcpy(str, result.c_str());
    return str;
}
int main() {
    char str[] = "Move-Hyphens-to-Front";
    cout << MoveHyphen(str, strlen(str)) << endl;
    return 0;
}

File: 009-Move Hyphens to Front/MoveHyphenstoFront.py
def MoveHyphen(s: str) -> str:
    if s is None:
        return None

    hyphen_part = ''
    other_part = ''

    for char in s:
        if char == '-':
            hyphen_part += '-'
        else:
            other_part += char

    return hyphen_part + other_part

# Sample usage
s = "Move-Hyphens-to-Front"
print(MoveHyphen(s))
